util/src/sync/spin.rs

```diff
@@ -14,13 +17,22 @@ pub struct MutexGuard<'a, T> {
 }
 
 impl<T> Mutex<T> {
+    #[cfg(any(test, feature = "loom"))]
     pub fn new(data: T) -> Self {
```

Comment:
I take it `loom` doesn't have a `const fn new` method? It'd be nice to keep the constness similar.

Reply:
> I take it `loom` doesn't have a `const fn new` method?

That's correct. Constructing a `loom` atomic or `CausalCell` inside of a test requires accessing the "execution" (test-run-specific data).

We can probably get rid of the `const fn` new actually, since the VGA buffer is in a `lazy_static`; I added this because I was hoping to use a regular static (but you can't do pointer casts in static initializers).